pkg/btree/multi.go

```diff
@@ -97,6 +99,60 @@ func (m *LinkedMetaPage) MarshalMetadata(bm encoding.BinaryMarshaler) error {
 	return m.SetMetadata(buf)
 }
 
+func (m *LinkedMetaPage) NextNOffsets() ([]uint64, error) {
+	offsets := make([]uint64, N)
+	currentOffset := m.offset
+
+	for i := 0; i < N; i++ {
+		if _, err := m.rws.Seek(int64(currentOffset)+12, io.SeekStart); err != nil {
+			return nil, err
+		}
+
+		var nextOffset uint64
+		if err := binary.Read(m.rws, binary.LittleEndian, &nextOffset); err != nil {
+			return nil, err
+		}
+
+		if nextOffset == ^uint64(0) {
+			break
+		}
+
+		offsets[i] = nextOffset
+		currentOffset = nextOffset
+	}
+
+	return offsets, nil
+}
+
+func (m *LinkedMetaPage) SetNextNOffsets() error {
+	offsets, err := m.NextNOffsets()
+
+	if err != nil {
+		return err
+	}
+
+	if len(offsets) > N {
+		return fmt.Errorf("too many offsets, max number of offsets should be %d", N)
+	}
+
+	if _, err := m.rws.Seek(int64(m.offset)+12, io.SeekStart); err != nil {
+		return err
+	}
+
+	for _, offset := range offsets {
+		if err := binary.Write(m.rws, binary.LittleEndian, offset); err != nil {
+			return err
+		}
+	}
+
+	for i := len(offsets); i < N; i++ {
+		if err := binary.Write(m.rws, binary.LittleEndian, ^uint64(0)); err != nil {
```

Comment:
Nice this is pretty clever. One improvement: since you stop iterating after the first ending offset, only one offset has to be set to 0xFFFFFFFF. The rest can be left as whatever's in the buffer.